src/pf-popover/pf-popover.component.js

```diff
@@ -0,0 +1,421 @@
+   */
+  get containerSelector() {
+    return this._containerSelector;
+  }
+
+  /**
+   * Set the popover containerSelector
+   *
+   * @param {string} value The container element selector
+   */
+  set containerSelector(value) {
+    if (this._containerSelector !== value) {
+      this._containerSelector = value;
+      this._container = document.querySelector(this._containerSelector);
+      this.setAttribute('containerSelector', value);
+    }
+  }
+
+  /**
+   * Get the animation duration
+   *
+   * @returns {string} The animation duration
+   */
+  get duration() {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation duration
+   *
+   * @param {string} value The animation duration
+   */
+  set duration(value) {
+    if (this._duration !== value) {
+      this._duration = value;
+      this.setAttribute('duration', value);
+    }
+  }
+
+  /**
+   * Get the animation delay
+   *
+   * @returns {string} The animation delay
+   */
+  get delay() {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation delay
+   *
+   * @param {string} value The animation delay
+   */
+  set delay(value) {
+    if (this._delay !== value) {
+      this._delay = value;
+      this.setAttribute('delay', value);
+    }
+  }
+
+  /**
+   * Get the placement this._placement
+   *
+   * @returns {string} The placement this._placement left, top, bottom, right
+   */
+  get placement() {
+    return this._placement;
+  }
+
+  /**
+   * Set placement this._placement
+   *
+   * @param {string} value The placement this._placement left, top, bottom, right
+   */
+  set placement(value) {
+    if (this._placement !== value) {
+      this._placement = value;
+      this.setAttribute('placement', value);
+    }
+  }
+
+  /**
+  * Get the targetSelector
+  *
+  * @returns {string} The target element selector
+  */
+  get targetSelector() {
+    return this._targetSelector;
+  }
+
+  /**
+   * Set targetSelector
+   *
+   * @param {string} value The target element selector
+   */
+  set targetSelector(value) {
+    if (this._targetSelector !== value) {
+      this._targetSelector = value;
+      this._target = document.querySelector(this._targetSelector);
+      this.setAttribute('targetSelector', value);
+    }
+  }
+
+  /**
+   * Get the popoverTitle
+   *
+   * @return {string} the title of popover
+   */
+  get popoverTitle() {
+    return this._popoverTitle;
+  }
+
+  /**
+   * Set popoverTitle
+   *
+   * @param {string} value The title of popover
+   */
+  set popoverTitle(value) {
+    if (this._popoverTitle !== value) {
+      this._popoverTitle = value;
+      this.setAttribute('popoverTitle', value);
+    }
+  }
+
+  /**
+     * The popover open method
+     */
+  open() {
+    clearTimeout(this._timer);
+    this._timer = setTimeout(() => {
+      if (this.popover === null) {
+        this._createPopover();
+        this._stylePopover();
+        this._checkPlacement();
+        this._showPopover();
```

Comment:
Found a couple of more things... this sequence is causing multiple nodes to be appended, it's repeated around ```line... 405``` little bit of DOM pollution...
```javascript
_checkPlacement() {
  if (!pfUtil.isElementInViewport(this.popover)) {
    this._placement = this._updatePlacement();
    this._createPopover();
    this._stylePopover();
  }
}
```

Both sequences are load order specific... this isn't a requirement but if you can think of a way to use state that would push this component pretty far